s3backgrounddelete/s3backgrounddelete/object_recovery_msgbus_consumer.py

```diff
@@ -0,0 +1,111 @@
+        pass
+
+    def _process_msg(self, msg):
+        """loads the json message and sends it to validation and processing"""
+        self._logger.info(
+            "Processing following records in consumer: " + msg)
+        try:
+            # msg: {"Key": "egZPBQAAAAA=-ZQIAAAAAJKc=",
+            #       "Value": "{\\"index_id\\":\\"egZPBQAAAHg=-YwIAAAAAJKc=\\",
+            #       \\"object_layout_id\\":1,
+            #        \\"object_metadata_path\\":\\"object1\\"}\\n"}
+            probable_delete_records = json.loads(msg)
+
+            if probable_delete_records:
+                validator = ObjectRecoveryValidator(
+                    self._config, probable_delete_records, self._logger)
+                validator.process_results()
+
+        except (KeyError, ValueError) as ex:    # Bad formatted message. Will discard it
+            self._logger.error("Failed to parse JSON data due to: " + str(ex))
+            return True
+        except Exception as ex:
+            self._logger.error(str(ex))
+            return False
+        return True
+
+    def receive_data(self):
+        """Initializes consumer, connects and receives messages from message bus"""
+        try:
+            self._consumer = S3CortxMsgBus()
+            ret,msg = self._consumer.connect()
+            if ret:
+                id = self._config.get_msgbus_consumer_id_prefix() + str(socket.gethostname())
+                group = self._config.get_msgbus_consumer_group()
+                type = self._config.get_msgbus_topic()
+                autoack = False
+                offset = 'earliest'
+                ret, msg = self._consumer.setup_receive(id, group, type, autoack, offset)
+                if ret:
+                    while True:
+                        # We will keep on receiving until there are messages to receive.
+                        # Failure is treated as no message in queue, in that case we sleep
+                        # for a specified duration and then try to receive again.
+                        # In case of non-daemon mode we will exit once we encounter failure
+                        # in receiving messages.
+                        ret,message = self._consumer.receive()
+                        if ret:
+                            # Process message can fail, but we still acknowledge the message
+                            # The last step in process message is to delete the entry from
+                            # probable delete index. Even if we acknowledge a message that
+                            # has failed being processed it would eventually come back as
+                            # the entry has not been deleted from probable delete index. 
+                            self._process_msg(msg.decode('utf-8'))
+                            self._consumer.commit()
+                        else:
+                            self._logger.error("Failed to receive msg from message bus : " + str(message))
```

Comment:
If there is no message, then it will fail, which is not error.
The logging need to be at  debug level

Reply:
Done